ImagePalette/SwiftPriorityQueue.swift

```diff
@@ -112,27 +112,37 @@ public struct PriorityQueue<T: Comparable> {
 }
 
 // MARK: - GeneratorType
-extension PriorityQueue: GeneratorType {
+extension PriorityQueue: IteratorProtocol {
 
     public typealias Element = T
     mutating public func next() -> Element? { return pop() }
 }
 
 // MARK: - SequenceType
-extension PriorityQueue: SequenceType {
+extension PriorityQueue: Sequence {
 
-    public typealias Generator = PriorityQueue
-    public func generate() -> Generator { return self }
+    public typealias Iterator = PriorityQueue
+    public func makeIterator() -> Iterator { return self }
 }
 
 // MARK: - CollectionType
-extension PriorityQueue: CollectionType {
+extension PriorityQueue: Collection {
 
     public typealias Index = Int
 
     public var startIndex: Int { return heap.startIndex }
     public var endIndex: Int { return heap.endIndex }
 
+    public func formIndex(after i: inout Int) {
+        if i + 1 < heap.endIndex {
+            i = i + 1
+        }
+    }
+
+    public func index(after i: Int) -> Int {
+        return i + 1
+    }
+
```

Comment:
the above 2 things I had to add to get rid of an error (conform to IndexableBase). seems right?